libyul/optimiser/StackToMemoryMover.cpp

```diff
@@ -77,17 +80,94 @@ void StackToMemoryMover::run(
 )
 {
 	VariableMemoryOffsetTracker memoryOffsetTracker(_reservedMemory, _memorySlots, _numRequiredSlots);
-	StackToMemoryMover stackToMemoryMover(_context, memoryOffsetTracker);
+	list<Statement> newFunctionDefinitions;
+	map<YulString, FunctionMoveInfo> functionMoveInfos;
```

Comment:
So this is pretty similar to the actual slot allocation in ``StackLimitEvader``.
We could also construct this map already there and pass it over here. It would make the ``fakeStackLimitEvader`` tests a bit more tricky, though (which is not a good reason not to do it, though).